common/persistence/sql/storage/mysql/visibility.go

```diff
@@ -106,6 +109,11 @@ func (mdb *DB) ReplaceIntoVisibility(row *sqldb.VisibilityRow) (sql.Result, erro
 	}
 }
 
+// DeleteFromVisibility deletes a row from visibility table if it exist
+func (mdb *DB) DeleteFromVisibility(filter *sqldb.VisibilityFilter) (sql.Result, error) {
+	return mdb.conn.Exec(templateDeleteWorkflowExecution, filter.DomainID, filter.RunID)
```

Comment:
Will this return error if target row is not found?

Reply:
it won't. Just the RowsAffected() will return 0